### PR TITLE
[FLINK-24292][connectors/kafka] Use KafkaSink in examples instead of …

### PR DESCRIPTION
**Pull request by fapaul**
…FlinkKafkaProducer

## What is the purpose of the change

Use KafkaSink in examples instead of FlinkKafkaProducer

## Verifying this change

I ran all the examples locally and they still work.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/statemachine/KafkaEventsGeneratorJob.java [hunk 1]
```diff
@@ -47,7 +49,16 @@ public static void main(String[] args) throws Exception {
...
 
         env.addSource(new EventsGeneratorSource(errorRate, sleep))
-                .addSink(new FlinkKafkaProducer<>(brokers, kafkaTopic, new EventDeSerializer()));
+                .sinkTo(
+                        KafkaSink.<Event>builder()
+                                .setBootstrapServers(brokers)
+                                .setRecordSerializer(
+                                        KafkaRecordSerializationSchema.builder()
+                                                .setValueSerializationSchema(
+                                                        new EventDeSerializer())
```
Comment (JingGe) [inline flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/statemachine/KafkaEventsGeneratorJob.java:58]: All other serialization classes are following the ***Schema naming convention. Maybe we should consider rename EventDeSerializer to EventSchema in a separate PR.

**Comment by JingGe** [comment]
LGTM

**Comment by fapaul** [comment]
@flinkbot run azure

**Comment by rmetzger** [comment]
The docs seem to be updated already.
Merging ...

**Pull request merged**